doc/howto/oidc_auth0.md

```diff
@@ -0,0 +1,77 @@
+(oidc-auth0)=
+# How to configure Auth0 as login method for the LXD UI
+
+Auth0 is a flexible, drop-in solution to add authentication and authorization services to your applications. Auth0 supports OIDC and can be used to authenticate users for the LXD UI. This guide shows you how to set up Auth0.com as the login method for the LXD UI.
+
+## Using Auth0.com to access the LXD UI
+
+1. Open a free account on [Auth0.com](https://auth0.com/).
+
+1. Once logged into the Auth0 web interface, from the main navigation's {guilabel}`Applications` section, select {guilabel}`Applications` > {guilabel}`Create application`.
+    - Use the default type of {guilabel}`Native` and click {guilabel}`Create`.
+
+1. Go to the {guilabel}`Settings` tab of your new application.
+    - Scroll to the {guilabel}`Allowed Callback URLs` field in this tab and enter your LXD UI address, followed by `/oidc/callback`.
+      - Example: `https://example.com:8443/oidc/callback`
+      - An IP address can be used instead of a domain name.
+      - Note `:8443` is the default listening port for the LXD server. It might differ for your setup. You can verify the LXD configuration value `core.https_address` to find the correct port for your LXD server. 
+    - Enable {guilabel}`Allow Refresh Token Rotation`.
+    - Scroll down to {guilabel}`Advanced Settings` and select the {guilabel}`Grant Types` tab. Enable {guilabel}`Device code`.
+    - Select {guilabel}`Save Changes`.
+
+1. Near the top of the {guilabel}`Settings` tab, locate the {guilabel}`Domain` field. Copy the value and add the `https://` prefix and the `/` suffix as in the example below. This is your OIDC issuer for LXD. Set this value in your LXD server configuration with the command:
+
+       lxc config set oidc.issuer=https://dev-example.us.auth0.com/
+
+1. From your Auth0 application's {guilabel}`Settings` tab, copy the {guilabel}`Client ID` and use it in your LXD server configuration:
+
+       lxc config set oidc.client.id=6f6f6f6f6f6f
+
+1. Finally, in the Auth0 interface's main navigation,  under {guilabel}`Applications`, select {guilabel}`Applications` > {guilabel}`APIs`. Copy the {guilabel}`API Audience` value, then use it as the OIDC audience in your LXD server configuration:
+
+       lxc config set oidc.audience=https://dev-example.us.auth0.com/api/v2/
+
+Now you can access the LXD UI with any browser and use single sign-on (SSO) login. Enter the credentials for Auth0.
+
+Users will have no permissions by default. You must set up permission groups in LXD to grant access to projects and instances. For connecting the permission group to a user you have two options:
+
+1. Map LXD permission groups to the user directly. The user object in LXD will only be created on the first login of that user to LXD.
+ 
+1. Configure roles in Aut0 and use automatic mapping to LXD groups as described below.
```

Comment:
Can you add links to relevant sections here please? There are sections on fine-grained authorization in `doc/explanation/authorization.md`